tools/lua_sstable_consumer.cc

```diff
@@ -0,0 +1,1003 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <fmt/chrono.h>
+#include <lua.hpp>
+#include <seastar/core/fstream.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/util/short_streams.hh>
+
+#include "reader_permit.hh"
+#include "schema.hh"
+#include "sstables/sstables.hh"
+#include "tools/lua_sstable_consumer.hh"
+#include "types/collection.hh"
+#include "types/tuple.hh"
+#include "utils/rjson.hh"
+
+// has to be below the utils/rjson.hh include
+#include <rapidjson/ostreamwrapper.h>
+
+namespace {
+
+static constexpr char partition_start_metatable[] = "partition_start";
+static constexpr char sstable_metatable[] = "sstable";
+static constexpr char managed_bytes_metatable[] = "managed_bytes";
```

Comment:
Do we need to expose managed_bytes? We should convert to regular lua types (we even have the code to do the conversion), or perhaps start with lua native byte strings. But managed_bytes itself is uninteresting to lua.

Reply:
Lua has just the following native data types: nil, boolean, number (secretly a separate integer and double), string and table. I too thought they have a bytes type as I recalled using it but turns out they don't and I was just using some C-type the API in that program was exposing, which is the same thing I do here. I thought about maybe using string, or just exposing `bytes` but that means I'd have to linearize everything when crossing to Lua land and we can have pretty large cells being passed to Lua. So I exposed managed bytes as-is (though only a tiny subset of its API). Maybe I can use a more generic `fragmented_bytes` name here. The `managed` part doesn't make any sense in the Lua script land.

Reply:
We are already able to expose all cql types, see lang/lua.cc.

We can expose both the correctly-typed value, and as raw bytes (lua/lang.cc uses lua_pushlstring()).

Reply:
I like this proposal very much, I don't know how I didn't think about this.
I was very thorn on how to expose cell values and collection keys to Lua. Currently we just convert everything to a human-readable string and expose this string. This is fine for most basic types, but the human-readable version of many types is very hard to work with (if you want to do filtering for example). This is why partitions and rows have a `key_raw` field as well, because I thought it would be too hard to try to filter partitions/rows based on the possibly very awkward human-readable representation (think frozen collections, or similar).
If we pushed all these as native types, than all this would be solved.